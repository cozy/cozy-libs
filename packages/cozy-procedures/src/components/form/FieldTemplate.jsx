import React from 'react'
import PropTypes from 'prop-types'
import { Label, translate } from 'cozy-ui/transpiled/react'

export const FieldTemplate = ({ children, label, id, t }) => (
  <div>
    {label && <Label htmlFor={id}>{t(label)}</Label>}
    {children}
  </div>
)

FieldTemplate.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  t: PropTypes.func.isRequired
}

export default translate()(FieldTemplate)
