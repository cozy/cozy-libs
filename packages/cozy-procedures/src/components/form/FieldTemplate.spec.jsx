import React from 'react'

import { FieldTemplate } from './FieldTemplate'
import tMock from '../../../test/__mocks__/tMock'

describe('FieldTemplate component', () => {
  it('should match snapshot', () => {
    const props = {
      id: '123',
      label: 'My field',
      t: tMock
    }
    const component = shallow(
      <FieldTemplate {...props}>
        <input name="foobar" />
      </FieldTemplate>
    )
    expect(component).toMatchSnapshot()
  })
})
