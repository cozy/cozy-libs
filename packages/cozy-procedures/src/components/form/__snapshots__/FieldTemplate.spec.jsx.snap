// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldTemplate component should match snapshot 1`] = `
<div>
  <Label
    block={true}
    className=""
    error={false}
    htmlFor="123"
  >
    My field 
  </Label>
  <input
    name="foobar"
  />
</div>
`;
